tools/smoke_test.py

```diff
@@ -0,0 +1,166 @@
+import os
+import re
+import subprocess
+import sys
+import traceback
+
+try:
+    import packaging.version
+except ImportError:
+    from pkg_resources import packaging
+Version = packaging.version.Version
+
+MIN_CUDA_VERSION = Version("11.6")
+MIN_PYTHON_VERSION = (3, 7)
+# NOTE: dev version required, e.g. Version('1.13.0.dev123') < Version('1.13.0')
+MIN_TORCH_VERSION = Version("1.13.0.dev0")
+
+
+def check_python():
+    if sys.version_info < MIN_PYTHON_VERSION:
+        raise RuntimeError(
+            f"Python version not supported: {sys.version_info} "
+            f"- minimum requirement: {MIN_PYTHON_VERSION}"
+        )
+    return sys.version_info
+
+
+# Checks for correct pip dependencies, according to `install_requires` in setup.py.
+def check_pip_deps():
+    proc = subprocess.run(["pip", "check"], capture_output=True)
+    if proc.returncode != 0:
+        dynamo_broken_deps = []
+        for line in proc.stdout.decode("utf-8").strip().split("\n"):
+            if line.startswith("torchdynamo"):
+                dynamo_broken_deps.append(line)
+        if len(dynamo_broken_deps) > 0:
+            raise RuntimeError(
+                "torchdynamo requirements not met:\n"
+                + "\n".join(map(str, dynamo_broken_deps))
+            )
+    proc = subprocess.run(["pip", "show", "torchdynamo"], capture_output=True)
+    if proc.returncode != 0:
+        raise RuntimeError("`torchdynamo` is not installed")
+
+
+# Checks for correct torch version.
+# Using check_pip_deps does not work if the minimum required torch version
+# is not present in PyPI.
+def check_torch():
+    import torch
+
+    if torch.__version__ < MIN_TORCH_VERSION:
+        raise RuntimeError(
+            f"torch version not supported: {torch.__version__} "
+            f"- minimum requirement: {MIN_TORCH_VERSION}"
+        )
+
+    return torch.__version__
+
+
+def check_cuda():
+    import torch
+
+    if not torch.cuda.is_available():
+        return None
+
+    torch_cuda_ver = Version(torch.version.cuda)
+    if torch_cuda_ver < MIN_CUDA_VERSION:
+        raise RuntimeError(
+            f"CUDA version not supported: {torch_cuda_ver} "
+            f"- minimum requirement: {MIN_CUDA_VERSION}"
+        )
+
+    # check if torch cuda version matches system cuda version
+    cuda_home = os.getenv("CUDA_HOME")
+    if not cuda_home:
+        raise RuntimeError("$CUDA_HOME not found")
+
+    nvcc = os.path.join(cuda_home, "bin", "nvcc")
+    nvcc_version_str = (
+        subprocess.check_output([nvcc, "--version"]).strip().decode("utf-8")
+    )
+    cuda_version = re.search(r"release (\d+[.]\d+)", nvcc_version_str)
```

Comment:
This feels assumptive, in order:

1) That nvcc is installed there (You could have a malformed $CUDA_HOME)
2) That running --version produces the string you want

Reply:
> I'm wary about calling this code since it's not public API, so I'm thinking of copying the parts I need.

Call it, its fine, its a private API in a repo we own, within the same overall project, for a niche tool. It certainly beats out copying the code.